### PR TITLE
Deadmin tweaks: Admins without +AUTOLOGIN start deadmined. AUTOLOGIN defaults to on.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Admins without `+AUTOLOGIN` start deadmined, reloading admins re-deadmins them.
This flag defaults to on, and instead is removed with `-AUTOLOGIN`
People with `+PERMISSION` can now deadmin or readmin other admins at will.

Deadmining no longer destroys the admin datum.
Moved some code around to make it more sane

The plan is to allow headmins to add admin candidates to the .txt as admin observers, who start deadmined and can re-admin while dead to watch ahelps or conflict resolution go down and learn even when no admin trainers are on to train them 

:cl:
server: Added new admin flag, AUTOLOGIN, to control if admins start with admin powers. this defaults to on, and can be removed with -AUTOLOGIN
admin: Admins with +PERMISSION may now deadmin or readmin other admins via the permission panel.
/:cl:

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
Good stuff

File: code/modules/admin/holder2.dm [hunk 1]
```diff
@@ -30,45 +35,55 @@ GLOBAL_PROTECT(href_token)
...
-/proc/HrefTokenFormField(forceGlobal = FALSE)
-	return "<input type='hidden' name='admin_token' value='[RawHrefToken(forceGlobal)]'>"
+	//only admins with +ADMIN start admined
+	if (force_active || (R.rights & R_ADMIN))
+		activate()
+	else
+		deactivate()
+
+
+/datum/admins/proc/activate()
```
Comment (kevinz000) [inline code/modules/admin/holder2.dm]: sleeper agents

File: code/modules/admin/admin_ranks.dm [hunk 2]
```diff
@@ -171,21 +171,19 @@ GLOBAL_PROTECT(admin_ranks)
 	#endif
 
 
-/proc/load_admins(target = null)
-	if(IsAdminAdvancedProcCall())
-		to_chat(usr, "<span class='admin prefix'>Admin Reload blocked: Advanced ProcCall detected.</span>")
-		return
```
Comment (MrStonedOne) [inline code/modules/admin/admin_ranks.dm]: This was removed because there is literally a verb given to all admins that calls this proc.

**Comment by JJRcop** [comment]
I think it is better to add a new admin right called +DEADMINNED, so that you can pick and choose which roles get initially deadminned

**Comment by MrStonedOne** [comment]
very like it would be the reverse, +AUTOLOGIN or some shit, as permission stuff assumes more flags == more power for certain admin to admin operation limiting. but +ADMIN is a very basic flag that almost every role has, people without it even get a special color in asay, that I thought it would be perfect canidate for a removal of auto-login.

I was initially going to make it so people without +ADMIN can't be logged in while their mob was a `/living/` but that turned out to require an unacceptable level of snowflake (Mainly because i wanted a way for any logged in admin to override it. This version of such a 

However such a change if it turned out be needed could be done easily with the change of 3 or so flag checks, so i'm worried about that.

**Comment by JJRcop** [comment]
The main reason if I was thinking people with the coder rank should have this so that they are normal people by default, then when trouble happens they can open their button up shirt and become CODER, THE DEBUGGER TO SAVE THE WORLD

**Comment by MrStonedOne** [comment]
But coders never play so thats not an issue. =P

**Comment by MrStonedOne** [comment]
@JJRcop done.

**Comment by optimumtact** [comment]
we play sometimes

**Comment by praisenarsie** [comment]
_sometimes_

**Pull request merged**